proxy/src/connection.rs

```diff
@@ -68,16 +78,19 @@ impl BoundPort {
     // This ensures that every incoming connection has the correct options set.
     // In the future it will also ensure that the connection is upgraded with
     // TLS when needed.
-    pub fn listen_and_fold<T, F, Fut>(self, executor: &Handle, initial: T, f: F)
-        -> Box<Future<Item = (), Error = io::Error> + 'static>
+    pub fn listen_and_fold<T, F, Fut>(self, handle: &Handle, initial: T, f: F)
```

Comment:
Why is `Handle` still needed here?

Reply:
Ah, thanks for catching that; this can probably just use `Handle::current()` :thumbsup: